packages/firebase_auth/firebase_auth_platform_interface/lib/src/platform_interface/platform_interface_firebase_auth.dart

```diff
@@ -76,6 +76,25 @@ abstract class FirebaseAuthPlatform extends PlatformInterface {
     _instance = instance;
   }
 
+  static Persistence? _persistence;
+
+  /// Changes the current type of persistence on the current Auth instance for
+  /// the currently saved Auth session and applies this type of persistence for
+  /// future sign-in requests, including sign-in with redirect requests.
+  ///
+  /// This makes it easy for a user signing in to specify whether their session
+  /// should be remembered or not. It also makes it easier to never persist the
+  /// Auth state for applications that are shared by other users or have
+  /// sensitive data.
+  ///
+  /// This is only supported on web based platforms.
+  // ignore: use_setters_to_change_properties
+  static void persistenceType(Persistence persistenceType) {
+    _persistence = persistenceType;
+  }
+
+  static Persistence? get persistence => _persistence;
```

Comment:
Should this not be stored on a per-instance basis rather than statics? The web sdk allows specifying per auth instance and not globally, e.g. I can have one auth persist and another not, so I'd imagine the internal constructor should take and store the setting instead here:

![image](https://user-images.githubusercontent.com/5347038/180477011-2ed16ff3-48cf-4d62-abc2-8ca897aa8f06.png)

 e.g. just pass it through to `FirebaseAuth._(app: app, persistence: persistence);`